File: python/src/shapes/base/commons/dimensional.py
from typing import Protocol


class Dimensional(Protocol):
    """
    # `class Dimensional(Protocol)`

    Defines the interface for measurable shapes.
    """

    def area(self) -> float:
        """
        Calculates the area of the shape.
        """
        return 0.0

    def perimeter(self) -> float:
        """
        Calculates the perimeter of the shape.
        """
        return 0.0

File: elixir/lib/shapes/base/commons/dimensional.ex
defprotocol Shapes.Base.Commons.Dimensional do
  @moduledoc """
  Protocol for measurable shapes.
  """

  @doc """
  Calculates the area of the shape.
  """
  def area(shape)

  @doc """
  Calculates the perimeter of the shape.
  """
  def perimeter(shape)
end
